### PR TITLE
Add compression algorithm selection support

### PR DESCRIPTION
**Pull request by Heavenser**
It look like this:

------------------------------
Bliss Initramfs - v7.1.3
Jonathan Vasquez <jon@xyinn.org>
2-BSD
------------------------------

Which initramfs features do you want? (Separated by a comma):

[>] 1. ZFS
[>] 2. LVM
[>] 3. RAID
[>] 4. LUKS
[>] 5. Basic

Features [1]: 1,4

Do you want to use the current kernel: 4.16.14-Hvs [Y/n]:

[*] Checking preliminary binaries ...
[*] Creating temporary directory at /home/heavenser/Downloads/bi-399972867 ...
[*] Checking required files ...
[+] Using LUKS
[+] Using ZFS
[*] Copying binaries ...
[*] Copying modules ...
[*] Generating modprobe information ...
[*] Copying library dependencies ...
[*] Creating symlinks ...
[*] Performing finishing steps ...

Which initramfs compress algorithm do you want? (You can choose only one):

[>] 1. GZIP
[>] 2. LZ4
[>] 3. XZ

Algorithm [1]: 2

[*] Creating the initramfs ...
77983 blocks
[*] Please copy "initrd-4.16.14-Hvs" to your /boot directory

### CODE REVIEW

File: pkg/libs/Core.py [hunk 1]
```diff
@@ -85,9 +85,9 @@ def PrintMenuAndGetDesiredFeatures(cls):
                 Raid.Enable()
             elif feature == "luks":
                 Luks.Enable()
-            # Just a base initramfs with no additional stuff
-            # This can be used with other options though
-            # (i.e you have your rootfs directly on top of LUKS)
+                # Just a base initramfs with no additional stuff
+                # This can be used with other options though
+                # (i.e you have your rootfs directly on top of LUKS)
```
Comment (fearedbliss) [inline pkg/libs/Core.py]: ?

File: pkg/libs/Core.py [hunk 2]
```diff
@@ -131,8 +131,8 @@ def GetDesiredKernel(cls):
 
                 if not var.kernel:
                     Tools.Fail("You didn't enter a kernel. Exiting...")
-            else:
-                Tools.Fail("Invalid Option. Exiting.")
+                else:
+                    Tools.Fail("Invalid Option. Exiting.")
```
Comment (fearedbliss) [inline pkg/libs/Core.py]: Is this suppose to be indented? Will lead to incorrect path if it wasn't.

File: pkg/libs/Core.py [hunk 3]
```diff
@@ -451,17 +451,48 @@ def CopyLibGccLibrary(cls):
...
+        else:
+            Tools.Warn("Exiting.")
+            quit(1)
+
+        return compress_cmd
+
     # Create the initramfs
     @classmethod
     def CreateInitramfs(cls):
+
```
Comment (fearedbliss) [inline pkg/libs/Core.py]: Printing and retrieicng should be done in the same function like get desired kernel and the print and get desired features function.

The function should be called in mkinitrd at the beginning of initramfs launch. Look at mkinitrd.py, specifically before

Core.VerifySupportedArchitecture()

File: pkg/libs/Core.py [hunk 4]
```diff
@@ -451,17 +451,48 @@ def CopyLibGccLibrary(cls):
...
         Tools.Info("Creating the initramfs ...")
 
         # The find command must use the `find .` and not `find ${T}`
         # because if not, then the initramfs layout will be prefixed with
         # the ${T} path.
         os.chdir(var.temp)
 
-        call(["find . -print0 | cpio -o --null --format=newc | gzip -9 > " + var.home + "/" + var.initrd], shell=True)
+        call(["find . -print0 | cpio -o --null --format=newc | " + cmp_cmd + " > " + var.home + "/" + var.initrd], shell=True)
```
Comment (fearedbliss) [inline pkg/libs/Core.py]: Once the above change is in affect, we don't need this local variable. Just read the one from var.cmdalgo directly.

**Comment by Heavenser** [comment]
Sorry, my code is dirty. I use a separate encrypted swap partition for hibernate, when I wake up my laptop from hibernation, the initramfs cannot resume data from swap partition, because it's not decrypted, so I have to add resume command to init script for resume data from a encrypted swap partition. Thanks.

**Pull request closed without merging**